File: src/schemas/audit.py
"""Audit log schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AuditLogRead(BaseModel):
    """Audit log entry returned by the admin query endpoint."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    actor_user_id: int | None
    action: str
    target_type: str
    target_id: int
    detail: str | None
    ip_address: str | None
    created_at: datetime


class CandidateActivityEvent(AuditLogRead):
    """Audit log entry enriched with job context for the candidate timeline.

    `job_title` is populated for `target_type="Application"` rows so the
    record pane can show which job an application event relates to.
    """

    job_title: str | None = None

File: frontend/src/types/audit.ts
/** Mirrors backend AuditLogRead schema. */
export interface AuditLogRead {
  id: number;
  actor_user_id: number | null;
  action: string;
  target_type: string;
  target_id: number;
  detail: string | null;
  ip_address: string | null;
  created_at: string;
}

/** Mirrors backend CandidateActivityEvent schema. */
export interface CandidateActivityEvent extends AuditLogRead {
  job_title: string | null;
}
